src/evm-block-extractor/src/storage_clients/gcp_big_query.rs

```diff
@@ -0,0 +1,263 @@
+use ethers_core::types::{Block, Transaction, TransactionReceipt, H256};
+use gcp_bigquery_client::model::query_request::QueryRequest;
+use gcp_bigquery_client::model::table_data_insert_all_request::TableDataInsertAllRequest;
+use gcp_bigquery_client::model::table_data_insert_all_request_rows::TableDataInsertAllRequestRows;
+use gcp_bigquery_client::Client;
+use serde::Serialize;
+
+use super::BlockChainDB;
+use crate::constants::{BLOCKS_TABLE_ID, PROJECT_ID, RECEIPTS_TABLE_ID};
+
+#[derive(Clone)]
+/// A client for BigQuery that can be used to query and insert data
+pub struct BigQueryBlockChain {
+    client: Client,
+    /// The project ID of the BigQuery project
+    project_id: String,
+    /// The dataset ID of the BigQuery table
+    ///
+    /// Can be mainnet/testnet
+    dataset_id: String,
+}
+
+impl BigQueryBlockChain {
+    /// Creates a new BigQuery client
+    /// The service account key should be stored in the
+    /// `GCP_BLOCK_EXTRACTOR_SA_KEY` environment variable
+    pub async fn new(dataset_id: String) -> anyhow::Result<Self> {
+        let sa_key = std::env::var("GCP_BLOCK_EXTRACTOR_SA_KEY")
```

Comment:
Clap can read default values from env variables too. Please, remove the `GCP_BLOCK_EXTRACTOR_SA_KEY` from here and create a proper entry in the clap configuration file for it.